From 825f988b917c3d9283b012a61d481fb9f6f85268 Mon Sep 17 00:00:00 2001
From: Ultra <ultra.public@proton.me>
Date: Thu, 19 Oct 2023 15:44:43 -0400
Subject: [PATCH] WIP: Add missing package dependencies

---
 zed_nodelets/CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/zed_nodelets/CMakeLists.txt b/zed_nodelets/CMakeLists.txt
index e23eed8..7b94f5b 100644
--- a/zed_nodelets/CMakeLists.txt
+++ b/zed_nodelets/CMakeLists.txt
@@ -30,6 +30,8 @@ endif()
 find_package(CUDA)
 checkPackage("CUDA" "CUDA not found, install it from:\n https://developer.nvidia.com/cuda-downloads")
 
+find_package(OpenCV)
+
 find_package(OpenMP)
 checkPackage("OpenMP" "OpenMP not found, please install it to improve performances: 'sudo apt install libomp-dev'")
 if (OPENMP_FOUND)
@@ -50,7 +52,7 @@ set(MODEL_LIB_DIR libs)
 set(YAML_INCLUDE ../depends/yaml-cpp/include)
 set(YAML_LIB_DIR ../depends/yaml-cpp/libs)
 
-include_directories(${MODEL_INCLUDE} ${YAML_INCLUDE})
+include_directories(${MODEL_INCLUDE} ${YAML_INCLUDE} /opt/tensorrt_inference/code/includes)
 link_directories(${MODEL_LIB_DIR} ${YAML_LIB_DIR})
 
 
@@ -67,7 +69,7 @@ find_package(catkin REQUIRED COMPONENTS
     nodelet
     tf2_geometry_msgs
     message_generation
-    diagnostic_updater    
+    diagnostic_updater
     dynamic_reconfigure
     zed_interfaces
 )
@@ -87,7 +89,7 @@ catkin_package(
     dynamic_reconfigure
     tf2_ros
     tf2_geometry_msgs
-    message_runtime    
+    message_runtime
     zed_interfaces
     geometry_msgs
     visualization_msgs
@@ -112,6 +114,7 @@ set(INCLUDE_DIRS
     ${catkin_INCLUDE_DIRS}
     ${CUDA_INCLUDE_DIRS}
     ${ZED_INCLUDE_DIRS}
+    ${OpenCV_INCLUDE_DIRS}
     ${CMAKE_CURRENT_SOURCE_DIR}/src/tools/include
     ${CMAKE_CURRENT_SOURCE_DIR}/src/zed_nodelet/include
     ${CMAKE_CURRENT_SOURCE_DIR}/src/rgbd_sensors_sync_nodelet/include
-- 
2.42.0

